tests/sepp/Workloads/GetByPrimaryKey.cpp

```diff
@@ -0,0 +1,181 @@
+    defaultThread.maxNumericKeyValue =
+        _options.defaultThreadOptions->maxNumericKeyValue;
+    defaultThread.fillBlockCache =
+        _options.defaultThreadOptions->fillBlockCache;
+    defaultThread.fetchFullDocument =
+        _options.defaultThreadOptions->fetchFullDocument;
+    defaultThread.collection = _options.defaultThreadOptions->collection;
+  }
+
+  WorkerThreadList result;
+  for (std::uint32_t i = 0; i < _options.threads; ++i) {
+    result.emplace_back(std::make_unique<Thread>(defaultThread, exec, server));
+  }
+  return result;
+}
+
+GetByPrimaryKey::Thread::Thread(ThreadOptions options, Execution& exec,
+                                Server& server)
+    : ExecutionThread(exec, server), _options(options) {}
+
+GetByPrimaryKey::Thread::~Thread() = default;
+
+void GetByPrimaryKey::Thread::run() {
+  // TODO - make more configurable
+
+  auto collection = _server.vocbase()->lookupCollection(_options.collection);
+  if (!collection) {
+    throw std::runtime_error("Could not find collection " +
+                             _options.collection);
+  }
+
+  RocksDBPrimaryIndex* primaryIndex = nullptr;
+  for (auto const& idx : collection->getIndexes()) {
+    if (idx->type() == arangodb::Index::TRI_IDX_TYPE_PRIMARY_INDEX) {
+      primaryIndex = static_cast<RocksDBPrimaryIndex*>(idx.get());
+      break;
+    }
+  }
+
+  if (!primaryIndex) {
+    throw std::runtime_error("Could not find primary index for collection " +
+                             _options.collection);
+  }
+
+  RocksDBCollection* rcoll =
+      static_cast<RocksDBCollection*>(collection->getPhysical());
+  std::uint64_t objectId = rcoll->objectId();
+  std::uint64_t indexId = primaryIndex->objectId();
+
+  rocksdb::ColumnFamilyHandle* docCF =
+      arangodb::RocksDBColumnFamilyManager::get(
+          arangodb::RocksDBColumnFamilyManager::Family::Documents);
+  rocksdb::ColumnFamilyHandle* indexCF =
+      arangodb::RocksDBColumnFamilyManager::get(
+          arangodb::RocksDBColumnFamilyManager::Family::PrimaryIndex);
+
+  auto& engine =
+      _server.vocbase()->server().getFeature<arangodb::RocksDBEngine>();
+  rocksdb::DB* rootDB = engine.db()->GetRootDB();
+
+  std::string temp;
+  RocksDBKey keyBuilder;
+  rocksdb::PinnableSlice val;
+
+  rocksdb::Status s;
+  rocksdb::ReadOptions ro(/*cksum*/ false, /*cache*/ _options.fillBlockCache);
+  ro.snapshot = rootDB->GetSnapshot();
+  auto guard = scopeGuard([&]() noexcept {
+    if (ro.snapshot) {
+      rootDB->ReleaseSnapshot(ro.snapshot);
+    }
+  });
+  ro.prefix_same_as_start = true;
+
+  for (std::uint64_t i = _options.minNumericKeyValue;
+       i < _options.maxNumericKeyValue; ++i) {
+    temp.clear();
+    if (!_options.keyPrefix.empty()) {
+      temp.append(_options.keyPrefix);
+    }
+    temp.append(std::to_string(i));
+
+    keyBuilder.constructPrimaryIndexValue(indexId, std::string_view(temp));
+
+    s = rootDB->Get(ro, indexCF, keyBuilder.string(), &val);
```

Comment:
Should we use `RocksDBPrimaryIndex::lookupKey` instead to avoid reimplementing this and instead rely on the same code that is used when running arangod?

Reply:
I internally did not use `lookupKey`, because it has a lot of code for consulting the in-memory cache etc.
In the SEPP tests I prefer to measure the storage engine performance and try to avoid most of the rest. Thus sticking to the RocksDB calls here.